src/geocat/viz/contourf/contourf.py

```diff
@@ -1,81 +1,435 @@
+
+            contourfontsize (:obj:`int`): Font size of the contour line labels. Default 12.
+
+            contourlabels (:obj:`list` or :class:`numpy.ndarray`): List or array of labels to use for contour line labels.
+
+            drawcontourlabels(:obj:`bool`): Whether add contour line labels to the figure.
+
+            flevels (:obj:`list` or :class:`numpy.ndarray`): List or array of levels to be passed into matplotlib's contourf function.
+
+            linecolor (:obj:`str`): Color of the contour line. Default "black".
+
+            linestyle (:obj:`str`): Linestyle of the contour line. Default solid for positive values, dashed for negative values.
+
+            linewidth (:obj:`int`): Width of the contour lines. Default 0.4.
+
+            manualcontourlabels (:obj:`bool`): Whether contour line labels should be manually drawn. Default False.
+
+            projection (:obj:`str`): Cartopy map projection. `See Cartopy documentation for full list. <https://scitools.org.uk/cartopy/docs/latest/crs/projections.html>`_
+
+            X (:class:`xarray.core.dataarray.DataArray'>` or :class:`numpy.ndarray`): The X axis data for the dataset. To be specified if not inferred correctly automatically.
+
+            Y (:class:`xarray.core.dataarray.DataArray'>` or :class:`numpy.ndarray`): The Y axis data for the dataset. To be specified if not inferred correctly automatically.
+        """
+
+        # Set default flevels, clevels, and colormap
         self._default_cmap = 'coolwarm'
         self._default_flevels = 5
         self._default_clevels = 7
 
-        # Pull out args
+        # Pull out args from data
         self.data = args[0]
 
-        # if in xarray, format as numpy array
+        # If xarray file, format as Numpy array
         if isinstance(self.data, xr.DataArray):
+            self.orig = self.data
             self.data = self.data.values
 
-        # Read in or calculate filled levels
-        if kwargs.get('flevels') is not 'auto':
-            # levels defined by kwargs
-            self.levels = kwargs.get('flevels')
-        elif kwargs.get('flevels') is not None:
-            # take a guess at filled levels
-            self._estimate_flevels
-
-        # Read in or calculate contour levels
-        if kwargs.get('clevels') is not 'auto':
-            # levels defined by kwargs
-            self.levels = kwargs.get('clevels')
-        elif kwargs.get('clevels') is not None:
-            # take a guess at filled levels
-            self._estimate_clevels
-
-        # Pull out child-class specific kwargs
+        # Read in or calculate filled levels using built in function
+        if kwargs.get('contour_fill') is not False:
+            if kwargs.get('flevels') is not None:
+                # levels defined by kwargs
+                self.levels = kwargs.get('flevels')
+            elif kwargs.get('flevels') is None:
+                # take a guess at filled levels
+                self._estimate_flevels
```

Comment:
```suggestion
                self._estimate_flevels()
```
I think you meant to call the internal function here. Line as written has no effect.